hail/src/main/scala/is/hail/cxx/StagedBaseStructBuilder.scala

```diff
@@ -3,45 +3,65 @@ package is.hail.cxx
 import is.hail.expr.types.physical.PBaseStruct
 import is.hail.utils.ArrayBuilder
 
-class StagedBaseStructBuilder(fb: FunctionBuilder, pStruct: PBaseStruct) {
-  val ab = new ArrayBuilder[Code]
-  val s = Variable("s", "char *", s"${ fb.getArg(1) }->allocate(${ pStruct.alignment }, ${ pStruct.byteSize })")
+class StagedBaseStructBuilder(pStruct: PBaseStruct, off: Expression) {
+
+  def setAllMissing(): Code =
+    s"memset($off, 0xff, ${ pStruct.nMissingBytes });"
+
+  def clearAllMissing(): Code =
+    s"memset($off, 0, ${ pStruct.nMissingBytes });"
+
+  def setMissing(idx: Int): Code = {
+    if (pStruct.fieldRequired(idx))
+      "abort();"
```

Comment:
I agree we shouldn't use abort.  Use the NativeStatus to return an interpretable error message?

Reply:
Oops, yeah. I had that there as a placeholder because I couldn't remember the syntax and then forgot to change it >.<